### ISSUE TITLE
jobs time out

### ISSUE THREAD

**Issue by bahmutov** [semantic-release/semantic-release#140]
I have a job tested on 3 node versions on travis and noticed it times out
Seems once the first job is successful, the remaining two are waiting for each other

The leader reports (105.1)

```
store build cache
0.00s6.47schanges detected, packing new archive
.
uploading archive
after_success.1
0.14s$ curl -Lo travis_after_all.py https://git.io/travis_after_all
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  3418  100  3418    0     0  26979      0 --:--:-- --:--:-- --:--:-- 26979
$ python travis_after_all.py
This is a leader
Leader waits for minions [u'105.2']...
Leader waits for minions [u'105.2']...
Leader waits for minions [u'105.2']...
Leader waits for minions [u'105.2']...
Leader waits for minions [u'105.2']...
Leader waits for minions [u'105.2']...
```

The second job reports (105.2)

```
store build cache
0.00s6.60schanges detected, packing new archive
.
uploading archive
after_success.1
0.17s$ curl -Lo travis_after_all.py https://git.io/travis_after_all
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  3418  100  3418    0     0  21013      0 --:--:-- --:--:-- --:--:-- 21013
after_success.2
0.08s$ python travis_after_all.py
This is a minion
after_success.3
0.00s$ export $(cat .to_export_back) &> /dev/null
$ npm run semantic-release
> check-more-types@0.0.0 semantic-release /home/travis/build/kensho/check-more-types
> semantic-release pre && npm publish && semantic-release post
[travis-after-all] Job succeeded (90158055d220971b8a6ae8bf75bcd18a2636ce32e982b9d47f71fff304a8c7f9)
[travis-after-all] Waiting for 105.1 and 105.3 to be done...
[travis-after-all] Waiting for 105.1 and 105.3 to be done...
[travis-after-all] Waiting for 105.1 to be done...
[travis-after-all] Waiting for 105.1 to be done...
[travis-after-all] Waiting for 105.1 to be done...
[travis-after-all] Waiting for 105.1 to be done...
[travis-after-all] Waiting for 105.1 to be done...
```

**Comment by bahmutov** [comment]
For now as a work around I am publishing to NPM only if running tests on a specific Node version using plugin https://github.com/bahmutov/condition-node-version

**Comment by hbetts** [comment]
@bahmutov so I ran into issues using travis_after_all as well. One issue, which is similar to the one you encountered, had two of my travis-ci jobs waiting on each other. Disabling, and re-enabling the travis build seemed to clear that up.

The other issue was that I wasn't using travis_after_all correctly. Though I had logs for a leader job, and minion jobs, my coverage and semantic-release scripts were called for all jobs. Turns out I needed to actually check various environmental variables to ensure that a job was actually the leader before executing post test tasks. My changes to correct this latter issue are here: https://github.com/hbetts/jenkins-coverage-badge/commit/ba6070c995ee4fedda67da4982a8aeaf9267cbcf
